leptos_dom/src/macro_helpers/into_style.rs

```diff
@@ -0,0 +1,209 @@
+            }
+        }
+    }
+}
+
+impl<T: IntoStyle> IntoStyle for (Scope, T) {
+    #[inline(always)]
+    fn into_style(self, _: Scope) -> Style {
+        self.1.into_style(self.0)
+    }
+}
+
+#[cfg(all(target_arch = "wasm32", feature = "web"))]
+#[doc(hidden)]
+#[inline(never)]
+pub fn style_helper(
+    el: &web_sys::Element,
+    name: Cow<'static, str>,
+    value: Style,
+) {
+    use leptos_reactive::create_render_effect;
+    use wasm_bindgen::JsCast;
+
+    let el = el.unchecked_ref::<web_sys::HtmlElement>();
+    let style_list = el.style();
+    match value {
+        Style::Fn(cx, f) => {
+            create_render_effect(cx, move |old| {
+                let mut new = f();
+                while let Style::Fn(_, f) = new {
+                    new = f();
+                }
+                let new = match new {
+                    Style::Value(value) => Some(value),
+                    Style::Option(value) => value,
+                    _ => unreachable!(),
+                };
+                if old.as_ref() != Some(&new) {
+                    style_expression(&style_list, &name, new.as_ref(), true)
+                }
+                new
+            });
+        }
+        Style::Value(value) => {
+            style_expression(&style_list, &name, Some(&value), false)
+        }
+        Style::Option(value) => {
+            style_expression(&style_list, &name, value.as_ref(), false)
+        }
+    };
+}
+
+#[cfg(all(target_arch = "wasm32", feature = "web"))]
+#[inline(never)]
+pub(crate) fn style_expression(
+    style_list: &web_sys::CssStyleDeclaration,
+    style_name: &str,
+    value: Option<&Cow<'static, str>>,
+    force: bool,
+) {
+    use crate::HydrationCtx;
+
+    if force || !HydrationCtx::is_hydrating() {
+        let style_name = wasm_bindgen::intern(style_name);
+
+        if let Some(value) = value {
+            if let Err(e) = style_list.set_property(style_name, &value) {
+                crate::error!("[HtmlElement::style()] {e:?}");
+            }
+        } else {
+            if let Err(e) = style_list.remove_property(style_name) {
+                crate::error!("[HtmlElement::style()] {e:?}");
+            }
+        }
+    }
+}
+
+macro_rules! style_type {
+    ($style_type:ty) => {
+        impl IntoStyle for $style_type {
+            fn into_style(self, _: Scope) -> Style {
+                Style::Value(self.to_string().into())
+            }
+        }
+
+        impl IntoStyle for Option<$style_type> {
+            fn into_style(self, _: Scope) -> Style {
+                Style::Option(self.map(|n| n.to_string().into()))
+            }
+        }
+    };
+}
+
+style_type!(&String);
+style_type!(usize);
+style_type!(u8);
+style_type!(u16);
+style_type!(u32);
+style_type!(u64);
+style_type!(u128);
+style_type!(isize);
+style_type!(i8);
+style_type!(i16);
+style_type!(i32);
+style_type!(i64);
+style_type!(i128);
+style_type!(f32);
+style_type!(f64);
+style_type!(char);
```

Comment:
I don't know anything yet. It's not appropriate to handle a bool style_type also?